File: scripts/generate_sprites.py
#!/usr/bin/env python3
"""
Generate simple game sprites for Zombie Rush game
Requires: pip install Pillow
"""

from PIL import Image, ImageDraw
import os

output_dir = os.path.join(os.path.dirname(__file__), '../public/assets/zombie')
os.makedirs(output_dir, exist_ok=True)

def create_player():
    """Create player sprite (blue character)"""
    img = Image.new('RGBA', (64, 64), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Body (light blue)
    draw.rectangle([10, 10, 54, 54], fill=(74, 144, 226), outline=(44, 90, 160), width=3)
    
    # Eyes
    draw.rectangle([20, 25, 28, 33], fill=(255, 255, 255))
    draw.rectangle([36, 25, 44, 33], fill=(255, 255, 255))
    
    # Mouth
    draw.rectangle([24, 40, 40, 44], fill=(255, 255, 255))
    
    img.save(os.path.join(output_dir, 'player.png'))
    print('✓ Created player.png')

def create_zombie():
    """Create zombie sprite (green character with red eyes)"""
    img = Image.new('RGBA', (64, 64), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Body (dark green)
    draw.rectangle([10, 10, 54, 54], fill=(107, 142, 35), outline=(74, 95, 26), width=3)
    
    # Red eyes
    draw.rectangle([18, 25, 26, 33], fill=(255, 0, 0))
    draw.rectangle([38, 25, 46, 33], fill=(255, 0, 0))
    
    # Blood mouth
    draw.rectangle([22, 38, 42, 46], fill=(139, 0, 0))
    
    # Teeth
    for i in range(4):
        draw.rectangle([24 + i*5, 38, 27 + i*5, 46], fill=(255, 255, 255))
    
    img.save(os.path.join(output_dir, 'zombie.png'))
    print('✓ Created zombie.png')

def create_bullet():
    """Create bullet sprite (yellow circle)"""
    img = Image.new('RGBA', (32, 32), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Yellow bullet
    draw.ellipse([4, 4, 28, 28], fill=(255, 215, 0), outline=(255, 165, 0), width=2)
    
    # Highlight
    draw.ellipse([8, 8, 16, 16], fill=(255, 255, 153))
    
    img.save(os.path.join(output_dir, 'bullet.png'))
    print('✓ Created bullet.png')

def create_background():
    """Create dark background texture"""
    import random
    img = Image.new('RGB', (800, 600), (26, 26, 26))
    draw = ImageDraw.Draw(img)
    
    # Add texture with random noise
    for _ in range(5000):
        x = random.randint(0, 799)
        y = random.randint(0, 599)
        gray = random.randint(0, 30)
        draw.rectangle([x, y, x+2, y+2], fill=(gray, gray, gray))
    
    # Add blood stains
    for _ in range(10):
        x = random.randint(0, 800)
        y = random.randint(0, 600)
        radius = random.randint(10, 40)
        draw.ellipse([x-radius, y-radius, x+radius, y+radius], 
                    fill=(139, 0, 0, 76))  # Semi-transparent
    
    img.save(os.path.join(output_dir, 'bg.png'))
    print('✓ Created bg.png')

if __name__ == '__main__':
    try:
        create_player()
        create_zombie()
        create_bullet()
        create_background()
        print('\n✅ All sprites generated successfully!')
    except ImportError:
        print('⚠️  Error: Pillow library not found. Install it with: pip install Pillow')
    except Exception as e:
        print(f'⚠️  Error: {e}')

File: scripts/generate-sprites.js
// Script to generate simple game sprites using Canvas
const fs = require('fs');
const { createCanvas } = require('canvas');
const path = require('path');

const outputDir = path.join(__dirname, '../public/assets/zombie');

// Ensure output directory exists
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

// Player sprite (blue square with face)
function createPlayerSprite() {
  const canvas = createCanvas(64, 64);
  const ctx = canvas.getContext('2d');
  
  // Body (light blue)
  ctx.fillStyle = '#4A90E2';
  ctx.fillRect(10, 10, 44, 44);
  
  // Outline
  ctx.strokeStyle = '#2C5AA0';
  ctx.lineWidth = 3;
  ctx.strokeRect(10, 10, 44, 44);
  
  // Eyes
  ctx.fillStyle = '#FFFFFF';
  ctx.fillRect(20, 25, 8, 8);
  ctx.fillRect(36, 25, 8, 8);
  
  // Mouth
  ctx.fillStyle = '#FFFFFF';
  ctx.fillRect(24, 40, 16, 4);
  
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(path.join(outputDir, 'player.png'), buffer);
  console.log('✓ Created player.png');
}

// Zombie sprite (green/red square with scary face)
function createZombieSprite() {
  const canvas = createCanvas(64, 64);
  const ctx = canvas.getContext('2d');
  
  // Body (dark green)
  ctx.fillStyle = '#6B8E23';
  ctx.fillRect(10, 10, 44, 44);
  
  // Outline
  ctx.strokeStyle = '#4A5F1A';
  ctx.lineWidth = 3;
  ctx.strokeRect(10, 10, 44, 44);
  
  // Red eyes
  ctx.fillStyle = '#FF0000';
  ctx.fillRect(18, 25, 8, 8);
  ctx.fillRect(38, 25, 8, 8);
  
  // Blood mouth
  ctx.fillStyle = '#8B0000';
  ctx.fillRect(22, 38, 20, 8);
  
  // Teeth
  ctx.fillStyle = '#FFFFFF';
  for (let i = 0; i < 4; i++) {
    ctx.fillRect(24 + i * 5, 38, 3, 8);
  }
  
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(path.join(outputDir, 'zombie.png'), buffer);
  console.log('✓ Created zombie.png');
}

// Bullet sprite (yellow circle)
function createBulletSprite() {
  const canvas = createCanvas(32, 32);
  const ctx = canvas.getContext('2d');
  
  // Yellow bullet
  ctx.fillStyle = '#FFD700';
  ctx.beginPath();
  ctx.arc(16, 16, 12, 0, Math.PI * 2);
  ctx.fill();
  
  // Outline
  ctx.strokeStyle = '#FFA500';
  ctx.lineWidth = 2;
  ctx.stroke();
  
  // Highlight
  ctx.fillStyle = '#FFFF99';
  ctx.beginPath();
  ctx.arc(12, 12, 4, 0, Math.PI * 2);
  ctx.fill();
  
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(path.join(outputDir, 'bullet.png'), buffer);
  console.log('✓ Created bullet.png');
}

// Background (dark grunge texture)
function createBackgroundSprite() {
  const canvas = createCanvas(800, 600);
  const ctx = canvas.getContext('2d');
  
  // Dark base
  ctx.fillStyle = '#1a1a1a';
  ctx.fillRect(0, 0, 800, 600);
  
  // Add texture with random noise
  for (let i = 0; i < 5000; i++) {
    const x = Math.random() * 800;
    const y = Math.random() * 600;
    const gray = Math.random() * 30;
    ctx.fillStyle = `rgb(${gray}, ${gray}, ${gray})`;
    ctx.fillRect(x, y, 2, 2);
  }
  
  // Add some blood stains
  ctx.fillStyle = 'rgba(139, 0, 0, 0.3)';
  for (let i = 0; i < 10; i++) {
    const x = Math.random() * 800;
    const y = Math.random() * 600;
    ctx.beginPath();
    ctx.arc(x, y, Math.random() * 30 + 10, 0, Math.PI * 2);
    ctx.fill();
  }
  
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(path.join(outputDir, 'bg.png'), buffer);
  console.log('✓ Created bg.png');
}

// Generate all sprites
try {
  createPlayerSprite();
  createZombieSprite();
  createBulletSprite();
  createBackgroundSprite();
  console.log('\n✅ All sprites generated successfully!');
} catch (error) {
  console.error('Error generating sprites:', error);
  console.log('\n⚠️  Note: Install canvas package: npm install canvas');
}

